qiskit/qobj/_validation.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""QObj validation module for validation against JSON schemas."""
+
+import json
+import os
+import logging
+import jsonschema
+
+from qiskit import QISKitError
+from qiskit import __path__ as qiskit_path
+
+_SCHEMAS = {}
+
+
+logger = logging.getLogger(__name__)
+
+
+def _load_qobj_schema():
+    """Loads the QObj schema for use in future validations."""
+    if 'qobj' not in _SCHEMAS:
```

Comment:
Would it make sense to make this function accept a parameter, and load a specific schema based on that parameter, instead of hard-coding the dict key? It feels a bit counter-intuitive to have the function load "all" the schemas (even if "all" currently means 1) and populate the dict unconditionally, making it a bit less dynamic than it will be needed as soon as more than one schema needs to be loaded. Might be something that is better tackled in future iterations?

Reply:
I believe this method should be added to a module in the directory above `qiskit/_validation.py`. It will be necessary for loading of all schemas in `qiskit/schemas`. I agree with @diego-plan9 it should accept a parameter specifying the schema rather than hardcoding the schema.

Reply:
I'm aware but did not want to generalize yet. I'm not sure if this should be a parametrized version of the function or a singleton object. Neither I'm sure if the dictionary should be indexed by name or by the schema path and linked to each Qobj class with an inner field. I think my preferred syntax would be something like:

```python
from schema import get_validator
get_validator().validate(qobj)
get_validator().validate(result)
get_validator().validate(backend_configuration)
```

And internally:

```python
# module schema
def get_validator():
  return _VALIDATOR_SINGLETON

class Validator():
  def validate(self, obj):
    schema = self._get_schema(obj.schema)
    jsonvalidator.validate(serialize_to_json(obj), schema)
```